msr-flows/msbr_compare/run_user_input.py

```diff
@@ -341,159 +433,378 @@ def cycle_time_decay(self, identifier='CTD'):
+            print('Running CTD')
+            CTD_identifier = 'CTD'
+            active_identifiers.append(CTD_identifier)
+            active_identifiers = list(set(active_identifiers))
+            builder = full_run_serp(
+                N_steps,
+                ui.base_material_path,
+                ui.template_path,
+                ui.template_name,
+                ui.start_time,
+                ui.end_time,
+                ui.list_inventory,
+                ui.element_flow_list,
+                output_path)
+            try:
+                builder.cycle_time_decay(identifier=CTD_identifier)
+            except Exception as e:
+                print(e)
+                print('CTD failed, removing from active identifiers.')
+                active_identifiers.remove(CTD_identifier)
+            end_timer_count = time.time()
+            print(f'Ran CTD, took {end_timer_count - start_timer_count}s')
+
+        if ui.cycle_rate:
+            start_timer_count = time.time()
+            print('Running CR')
+            CR_identifier = 'CR'
+            active_identifiers.append(CR_identifier)
+            active_identifiers = list(set(active_identifiers))
+            builder = full_run_serp(
+                N_steps,
+                ui.base_material_path,
+                ui.template_path,
+                ui.template_name,
+                ui.start_time,
+                ui.end_time,
+                ui.list_inventory,
+                ui.element_flow_list,
+                output_path)
+            try:
+                builder.cycle_rate(identifier=CR_identifier)
+            except Exception as e:
+                print(e)
+                print('CR failed, removing from active identifiers.')
+                active_identifiers.remove(CR_identifier)
+            end_timer_count = time.time()
+            print(f'Ran CR, took {end_timer_count - start_timer_count}s')
+
+        if ui.saltproc_cycle_rate:
+            start_timer_count = time.time()
+            print('Running SPCR')
+            SPCR_identifier = 'SPCR'
+            active_identifiers.append(SPCR_identifier)
+            active_identifiers = list(set(active_identifiers))
+            builder = full_run_serp(
+                N_steps,
+                ui.base_material_path,
+                ui.template_path,
+                ui.template_name,
+                ui.start_time,
+                ui.end_time,
+                ui.list_inventory,
+                ui.element_flow_list,
+                output_path)
+            try:
+                builder.SP_cycle_rate(identifier=SPCR_identifier)
+            except Exception as e:
+                print(e)
+                print('SPCR failed, removing from active identifiers.')
+                active_identifiers.remove(SPCR_identifier)
+            end_timer_count = time.time()
+            print(f'Ran SPCR, took {end_timer_count - start_timer_count}s')
+
+        # Plotting
+
+        if ui.model_plotting:
```

Comment:
The following if statements could be one function that calls different methods for plotting. Avoid unnecessary code repetitions.

Reply:
Created functions for the different plotting approaches, will make debugging easier as well. Thank you for commenting on this.